Debug Shared Runtime deployment attempts are crashing on startup

### Steps to Reproduce

1. Create a new or load an existing project and deploy to device in Debug mode.

<!--
If you have a repro project, you may drag & drop the .zip/etc. onto the issue editor to attach it.
-->


### Actual Behavior
```
05-08 10:51:06.524  7834  7834 E mono    : Unhandled Exception:
05-08 10:51:06.524  7834  7834 E mono    : System.ArgumentException: Handle must be valid.
05-08 10:51:06.524  7834  7834 E mono    : Parameter name: instance
05-08 10:51:06.524  7834  7834 E mono    :   at Java.Interop.JniEnvironment+InstanceMethods.CallObjectMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method) [0x00009] in <872818d1526e40d0b1926facf75f539d>:0
```
### Version Information
XA 9.4.0.4 (HEAD/cc0e8b8d5)
<!--
1. On macOS and within Visual Studio, select Visual Studio > About Visual Studio, then click the Show Details button, then click the Copy Information button.

2. Paste below this comment block.
-->

### Log File
```
05-06 15:44:39.605 W/ActivityManager( 1516): Activity pause timeout for ActivityRecord{5104222 u0 Notepad.Notepad/md57135d3c68dd99636a6eb42c8fba6d00e.NotesListActivity t75}
05-06 15:44:39.605 V/ActivityManager( 1516): Scheduling idle now: forceIdle=trueimmediate=true
05-06 15:44:39.632 D/Mono    (25698): Loading reference 7 of Mono.Android.dll asmctx DEFAULT, looking for System.Runtime.Serialization, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
05-06 15:44:39.633 D/Mono    (25698): Image addref System.Runtime.Serialization[0x76711b7880] (asmctx DEFAULT) -> System.Runtime.Serialization.dll[0x765f2e5800]: 2
05-06 15:44:39.633 D/Mono    (25698): Prepared to set up assembly 'System.Runtime.Serialization' (System.Runtime.Serialization.dll)
05-06 15:44:39.633 D/Mono    (25698): Assembly System.Runtime.Serialization[0x76711b7880] added to domain RootDomain, ref_count=1
05-06 15:44:39.634 D/Mono    (25698): AOT: image 'System.Runtime.Serialization.dll.so' not found: (null)
05-06 15:44:39.634 D/Mono    (25698): AOT: image '/Users/builder/jenkins/workspace/archive-mono/2018-10/android/release/sdks/out/android-arm64-v8a-release/lib/mono/aot-cache/arm64/System.Runtime.Serialization.dll.so' not found: (null)
05-06 15:44:39.634 D/Mono    (25698): Config attempting to parse: 'System.Runtime.Serialization.dll.config'.
05-06 15:44:39.634 D/Mono    (25698): Config attempting to parse: '/Users/builder/jenkins/workspace/archive-mono/2018-10/android/release/sdks/out/android-arm64-v8a-release/etc/mono/assemblies/System.Runtime.Serialization/System.Runtime.Serialization.config'.
05-06 15:44:39.634 D/Mono    (25698): Assembly Ref addref Mono.Android[0x76711b7000] -> System.Runtime.Serialization[0x76711b7880]: 2
05-06 15:44:39.635 D/Mono    (25698): Loading reference 0 of System.Runtime.Serialization.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
05-06 15:44:39.635 D/Mono    (25698): Assembly Ref addref System.Runtime.Serialization[0x76711b7880] -> mscorlib[0x767988fc80]: 6
05-06 15:44:39.638 E/mono    (25698): 
05-06 15:44:39.638 E/mono    (25698): Unhandled Exception:
05-06 15:44:39.638 E/mono    (25698): System.ArgumentException: Handle must be valid.
05-06 15:44:39.638 E/mono    (25698): Parameter name: instance
05-06 15:44:39.638 E/mono    (25698):   at Java.Interop.JniEnvironment+InstanceMethods.CallObjectMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method) [0x00009] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Android.Runtime.JNIEnv.CallObjectMethod (System.IntPtr jobject, System.IntPtr jmethod) [0x0000e] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Interop.TypeManager.GetClassName (System.IntPtr class_ptr) [0x00000] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00054] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x000c1] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Java.Lang.Thread.get_DefaultUncaughtExceptionHandler () [0x0001e] in :0 
05-06 15:44:39.638 E/mono    (25698):   at Android.Runtime.JNIEnv.Initialize (Android.Runtime.JnienvInitializeArgs* args) [0x00202] in :0 
05-06 15:44:39.639 E/mono-rt (25698): [ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: Handle must be valid.
05-06 15:44:39.639 E/mono-rt (25698): Parameter name: instance
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Interop.JniEnvironment+InstanceMethods.CallObjectMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method) [0x00009] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Android.Runtime.JNIEnv.CallObjectMethod (System.IntPtr jobject, System.IntPtr jmethod) [0x0000e] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Interop.TypeManager.GetClassName (System.IntPtr class_ptr) [0x00000] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00054] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x000c1] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Java.Lang.Thread.get_DefaultUncaughtExceptionHandler () [0x0001e] in :0 
05-06 15:44:39.639 E/mono-rt (25698):   at Android.Runtime.JNIEnv.Initialize (Android.Runtime.JnienvInitializeArgs* args) [0x00202] in :0 
05-06 15:44:39.677 I/ActivityManager( 1516): Process Notepad.Notepad (pid 25698) has died: fore TPSL(308,3204)
```
<!--
1. On macOS and within Visual Studio:
    a. Click **Tools** > **SDK Command Prompt**.
    b. Within the launched `Terminal.app` window, run:

            adb logcat -d | pbcopy

2. Paste below this comment block
-->



<!--
Switch to the "Preview" tab to ensure your issue renders correctly.

Please add an appropriate "Area:" label in the Labels section.
-->

--- a/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs
@@ -308,10 +308,8 @@ namespace Xamarin.Android.Tasks
 		void WriteTypeMappings (List<TypeDefinition> types)
 		{
 			void logger (TraceLevel level, string value) => Log.LogDebugMessage (value);
-			TypeNameMapGenerator createTypeMapGenerator () => UseSharedRuntime ?
-				new TypeNameMapGenerator (types, logger) :
-				new TypeNameMapGenerator (ResolvedAssemblies.Select (p => p.ItemSpec), logger);
-			using (var gen = createTypeMapGenerator ()) {
+			TypeNameMapGenerator generator = new TypeNameMapGenerator (ResolvedAssemblies.Select (p => p.ItemSpec), logger);
+			using (var gen = generator) {
 				using (var ms = new MemoryStream ()) {
 					UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), "jm", ms, gen.WriteJavaToManaged);
 					UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), "mj", ms, gen.WriteManagedToJava);

